big_tests/tests/push_integration_SUITE.erl

```diff
@@ -264,26 +277,38 @@ inbox_msg_reset_unread_count(Config, Service, EnableOpts) ->
 
 
 muclight_inbox_msg_unread_count(Config, Service, EnableOpts) ->
-    escalus:story(
+    escalus:fresh_story(
       Config, [{alice, 1}, {kate, 1}, {bob, 1}],
       fun(Alice, Kate, Bob) ->
-              Room = room_name(Config),
-              RoomInfo = create_room(Room, [alice, kate, bob], Config),
+              Room = fresh_room_name(),
+              RoomJID = muc_light_helper:given_muc_light_room(Room, Alice, []),
+              KateJid = inbox_helper:to_bare_lower(Kate),
+
+              when_muc_light_affiliations_are_set(Alice, Room, [{Kate, member}]),
+              muc_light_helper:verify_aff_bcast([{Kate, member}, {Alice, owner}], [{Kate, member}]),
+
               KateToken = enable_push_for_user(Kate, Service, EnableOpts),
-              BobToken = enable_push_for_user(Bob, Service, EnableOpts),
 
-              send_message_to_room(Alice, Room),
-              check_notification(KateToken, 1),
+              SenderJID = muclight_conversation(Alice, RoomJID, <<"First!">>),
+              Notification = wait_for_push_request(KateToken),
+              assert_push_notification(Notification, Service, EnableOpts, SenderJID,
+                                       [{body, <<"First!">>}, {unread_count, 1}, {badge, 1}]),
 
-              send_message_to_room(Alice, Room),
-              check_notification(KateToken, 2),
+              muclight_conversation(Alice, RoomJID, <<"Second!">>),
+              escalus:wait_for_stanza(Alice),
+
+              Notification2 = wait_for_push_request(KateToken),
+              assert_push_notification(Notification2, Service, EnableOpts, SenderJID,
+                                       [{body, <<"Second!">>}, {unread_count, 2}, {badge, 1}]),
+
+              timer:sleep(1000),
```

Comment:
Why this sleep? What are we waiting for here? We could instead use one of those `mongoose_helper:wait_until` and check periodically for the condition we need.

Reply:
Ah, check https://github.com/esl/MongooseIM/pull/2394/commits/737f4eae581dbff5ec7e9c0a3a967f1f8f9d9bf8 :)